crates/ruff_formatter/src/printer/mod.rs

```diff
@@ -143,7 +163,13 @@ impl<'a> Printer<'a> {
 
             FormatElement::SourcePosition(position) => {
                 self.state.source_position = *position;
-                self.push_marker();
+                // The printer defers printing indents until the next text
+                // is printed. Pushing the marker now would mean that the
+                // mapped range includes the indent range, which we don't want.
+                // Only add a marker if we're not in an indented context, e.g. at the end of the file.
+                if self.state.pending_indent.is_empty() {
+                    self.push_marker();
+                }
```

Comment:
This change fixes the source map offsets for nodes. 

We emit a source map entry at the start of every node. For example:

```python
def a():
    pass
```

We emit a source map entry at the start of `def a` (0) and another for `pass` (13). 

**Before**:

The source map entry started at position 9 (before the indentation) which is unexpected because it is wider than the node's range.

**After**:

The source map begins after the indent at position 13.